site/content/blog/release-0.30.0.md

```diff
@@ -0,0 +1,174 @@
+13. Remove useless conversions `LockTime` -> `LockTime` (clippy has a lint for it)
+
+## Re-exports
+
+We are trying to separate the API from the directory structure, as part of this we are attempting to
+only re-export from the crate root types that exist in the standard Bitcoin vernacular i.e., for
+pubkey you can `use bitcoin::PublicKey` but to get an x-only pubkey you need to go to the `key`
+module `use bitcoin::key::XOnlyPublicKey`.
+
+Please note this is still work-in-progress, suggestions welcome.
+
+## Code moves
+
+We moved _a lot_ of stuff around. This was a precursor to crate smashing which we have now started.
+Hopefully things are in intuitive places, it might be useful to take a quick look at the new
+module structure to get a feel for things, specifically:
+
+- We have a workspace now! The main crate now lives in `bitcoin/`. The `bitcoin_hashes` repository
+  has been merged into the `rust-bitcoin` repository and now lives under `hashes/`.
+
+- The `util` module is all but gone, try just removing `util::` at first, most modules are
+  re-exported at the crate root.
+
+- Cryptography related stuff is now in the `crypto` module, submodules (excl. `taproot`) are
+  re-exported so you can import them directly (e.g. `use bitcoin::ecdsa`). We re-export the
+  `taproot` crypto stuff from the `taproot` module at the crate root so its usage is uniform with
+  the other `crypto` submodules. Modules include: (`ecdsa`, `taproot`, `sighash`, `key`).
+
+- Some hash wrapper types are now to be found in the module that they are used in e.g.,
+  `TapLeafHash` is in the `taproot` module. Others are still in `hash_types` but the re-exports now
+  conform to the aim stated above so you might need to add `hash_types::` to your paths for the more
+  esoteric hash types.
+
+## Script changes
+
+To optimize efficiency of working with borrowed scripts we renamed `Script` to `ScriptBuf` and added
+an unsized `Script`. It works just like `PathBuf` and `Path` (and other such types) from `std`. The
+API tries to resemble those types as much as reasonable (deref coercions etc.), so it should be
+intuitive. Methods in the library that previously took `&Script` (which is now `ScriptBuf`) take the
+unsized `&Script` now.
+
+Additionally, we changed the type accepted by the `push_slice` method to be another unsized newtype:
+`PushBytes` and it's owned counterpart `PushBytesBuf`. These types maintain the invariant of storing
+at most 2^32-1 bytes - the maximum one can push into script. Previously the method would panic if
+you attempted to do it (and it wasn't documented, sorry about that). Now you can either handle it
+explicitly using `TryFrom` or just pass a known-length array (implemented for arrays up to 73
+bytes).
+
+Types that are commonly pushed into script (serialized signatures, public keys...) implement
+`AsRef<PushBytes>` so you can pass those directly as well. You can also implement `AsRef` for your
+types so they can be pushed directly.
+
```

Comment:
I think taproot changes deserve their own section

> ## Taproot changes
>
> Since the introduction of Taproot support the API got "stress tested" and various issues were uncovered and fixed.
> One of them is that the name "Branch Hash" refers to the *algorithm* being used to compute the value but the resulting value is actually *a node* in the tree. Thus we renamed `TapBranchHash` to `TapNodeHash` to reflect this. Additionally we originally used raw `sha256::Hash`es for node hashes which was error-prone and annoying because of manual conversions. We changed them to use `TapNodeHash` instead.
>
> When writing smart contracts that have their taproot trees statically known it was annoying to use `TaprootMerkleBranch::try_from` for arrays that are statically known to be shorter than 128. We've added `From` conversion for these.
>
> `NodeInfo` got some improvements too. For trees that are statically guaranteed to not have hidden nodes we have `TapTree` type and both got methods for getting an iterator over their respective items. This also improved our code internally, fixing some bugs and decreasing the risk of other bugs. We have some additional ideas to improve it further.
>
> Some types that do not have a well-defined serialization in Bitcoin ecosystem or are purely Rust constructs (e.g. builders) got `serde` support removed. Serializing these would be error-prone and difficult to support stably. You can instead (de)serialize other types and convert them.
>
> Overall, these changes should make working with Taproot less error-prone and more ergonomic. Taproot is still young technology so it's possible there will be more changes in the future as new users try out the API. Please let us know if you have questions or suggestions.

cc @sanket1729 please check at least the part about `NodeInfo` if you can and LMK if it should be improved.

Reply:
Used as is pending comments from @sanket1729 (one or two minor grammar changes).

Reply:
@Kixunil, thanks for writing these down. ACK these changes.